syntax = "proto3";
package ortools_vrp;

message Matrix {
  repeated uint32 data = 2 [packed=true];
}

message TimeWindow {
  int64 start = 1;
  int64 end = 2;
  uint32 late_multiplier = 3;
}

message Service {
  repeated TimeWindow time_windows = 1;
  repeated uint32 quantities = 2;
  uint32 duration = 3;
  repeated uint32 vehicle_indices = 4;
}

message Rest {
  repeated TimeWindow time_windows = 1;
  uint64 duration = 2;
}

message Capacity {
  int32 limit = 1;
  float overload_multiplier = 2;
}

message Vehicle {
  Matrix time_matrix = 1;
  Matrix distance_matrix = 2;
  repeated Capacity capacities = 3;
  TimeWindow time_window = 4;
  repeated Rest rests = 5;
}

message Problem {
  repeated Vehicle vehicles = 3;
  repeated Service services = 4;
}
